infrastructure/code/template/details.html

```diff
@@ -542,22 +605,101 @@ <h2 class="example-fixes-header">External example fixes</h2>
+    const EXAMPLE_TAB_NAV_SELECTOR = '.example-fixes-tabs-nav';
+    const EXAMPLE_TAB_ITEM_SELECTOR = '.example-fixes-tabs-nav .tab-item';
+    const EXAMPLE_TAB_CONTENT_SELECTOR = '.example-fixes .tab-content';
+    const IGNORE_WARNING_SELECTOR = '.ignore-warning-wrapper';
+    const FIRST_EXAMPLE_CONTENT_ID = '#tab-content-0';
+    const ISSUE_OVERVIEW_MD_HEADER = '.issue-overview .overview-text h2';
+
+    document.addEventListener('DOMContentLoaded', () => {
+      const overviewHeader = document.querySelector(ISSUE_OVERVIEW_MD_HEADER);
+      if (overviewHeader) {
+        overviewHeader.classList.add('issue-overview-header');
+      }
+
+      const mainTabLinks = document.querySelectorAll(MAIN_TAB_ITEM_SELECTOR);
+      const mainTabContents = document.querySelectorAll(MAIN_TAB_CONTENT_SELECTOR);
+
+      const isIgnored = document.querySelector(IGNORE_WARNING_SELECTOR);
+
+      // On load, set the last selected tab and content
+      let lastSelectedTab = isIgnored ? 'ignore-details' : 'fix-analysis';
+      let lastSelectedFixContent = document.querySelector(FIRST_EXAMPLE_CONTENT_ID);
+
+      // Ensure the default nested content is displayed
+      if (lastSelectedFixContent) {
+        lastSelectedFixContent.classList.add('is-selected');
+      }
+
+      document.querySelector(MAIN_TAB_NAV_SELECTOR).addEventListener('click', (event) => {
+        const clickedTab = event.target.closest('.tab-item');
+        if (!clickedTab) return;
+
+        const selectedTab = clickedTab.getAttribute('data-tab');
+
+        // Save the current selected tab
+        lastSelectedTab = selectedTab;
+
+        // Toggle selected tab and content
+        mainTabLinks.forEach(tab => {
+          tab.classList.toggle('is-selected', tab === clickedTab);
+        });
+
+        mainTabContents.forEach(content => {
+          content.classList.toggle('is-selected', content.getAttribute('data-content') === selectedTab);
+        });
+
+        // Ensure the last selected nested tab content is displayed when switching back
+        if (selectedTab === 'fix-analysis' && lastSelectedFixContent) {
+          document.querySelectorAll(EXAMPLE_TAB_CONTENT_SELECTOR).forEach(content => {
+            content.classList.remove('is-selected');
+          });
+          lastSelectedFixContent.classList.add('is-selected');
+        }
       });
 
-      const targetContentId = clickedTab.id.replace('tab-link', 'tab-content');
-      tabContents.forEach(content => {
-        content.classList.toggle('is-selected', content.id === targetContentId);
+      // Restore the last selected tab on page load
+      mainTabContents.forEach(content => {
+        content.classList.toggle('is-selected', content.getAttribute('data-content') === lastSelectedTab);
       });
-    });
 
+      mainTabLinks.forEach(tab => {
+        tab.classList.toggle('is-selected', tab.getAttribute('data-tab') === lastSelectedTab);
+      });
```

Comment:
I just tested it and the old UI doesn't support this, as far as I can tell. What's the reason for adding this logic? I think it makes it a bit harder to understand so if you agree that the old UI doesn't have this, then we should remove it just to make our lives easier

Reply:
Actually, maybe I didn't understand what this does. Is it there to just make sure we don't end up with empty content under the example fix tab? 
![Screenshot 2024-05-31 at 17 41 59](https://github.com/snyk/snyk-ls/assets/81559517/4ca54eec-42f7-40c6-b811-0a3edec7a07f)

Either way it might be worth adding a comment here 😄

Reply:
I have refactored a bit more the code and added some comments. As I've shown before, I needed to find a way to keep the state of the latest content shown for the Code Diff suggestion and also to have it selected on page load.

This logic does not exist in the current VSCode implementation because we do not have tabs within tabs structure. Eventually we will replace the tabs for the Code Examples with the arrows.